marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/step/getStepsByType.sjs

```diff
@@ -0,0 +1,29 @@
+/**
+ Copyright (c) 2020 MarkLogic Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+'use strict';
+
+const Artifacts = require('/data-hub/5/artifacts/core.sjs');
+
+var propertiesToReturn;
+
+const response = {};
+const stepTypes = ['ingestion', 'mapping'];
+
+for (const stepType of stepTypes) {
```

Comment:
Because this is the only client that needs to only return certain properties for each step, I'd rather that logic be in here as opposed to being in artifacts/core.sjs. Once we have a second use case for this logic, we can consider how to reuse the logic. 

Also, since this endpoint is only used for returning stepId and name, I think that logic should go in here as well. If we later have a use case for getting all properties of steps or a different set, we can consider how to address it. 

In short, I'd postpone adding flexibility until we need it, because that flexibility comes with a cost.